ducktape/cluster/remoteaccount.py

```diff
@@ -12,15 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
 import os
-import select
+from paramiko import SSHClient, SSHConfig, AutoAddPolicy
+import shutil
 import signal
-import subprocess
+import socket
+import stat
 import tempfile
 from contextlib import contextmanager
 
 from ducktape.utils.http_utils import HttpMixin
 from ducktape.utils.util import wait_until
+from ducktape.errors import DucktapeError
+
+
+class RemoteAccountError(DucktapeError):
+    """This exception is raised when an attempted action on a remote node fails.
+    """
+    def __init__(self, account, msg):
+        self.account_str = str(account)
+        self.msg = msg
+
+    def __str__(self):
+        return "%s: %s" % (self.account_str, self.msg)
+
+
+class RemoteCommandError(RemoteAccountError):
+    """This exception is raised when a process run by ssh*() returns a non-zero exit status.
+    """
+    def __init__(self, account, cmd, exit_status, remote_err_msg):
```

Comment:
nit: `remote_err_msg` is pretty verbose where `msg` conveys basically the same info given the context